src/ChunkData.cpp

```diff
@@ -46,103 +39,90 @@ cChunkData::cChunkData(cAllocationPool<cChunkData::sChunkSection> & a_Pool) :
 
 cChunkData::~cChunkData()
 {
-	#if __cplusplus < 201103L
-		// auto_ptr style interface for memory management
-		if (!m_IsOwner)
-		{
-			return;
-		}
-	#endif
-	for (size_t i = 0; i < NumSections; i++)
-	{
-		Free(m_Sections[i]);
-		m_Sections[i] = nullptr;
-	}
+	Clear();
 }
 
 
 
 
 
-#if __cplusplus < 201103L
-	// auto_ptr style interface for memory management
-	cChunkData::cChunkData(const cChunkData & a_Other) :
-		m_IsOwner(true),
-		m_Pool(a_Other.m_Pool)
+cChunkData::cChunkData(const cChunkData & a_Other):
```

Comment:
@madmaxoft But we never are, because predicting/handling out of memory on a consumer OS is basically impossible. The only way I am aware of to handle it in a sane way is a configurable application limit on pool size.